app/models/report.rb

```diff
@@ -51,6 +51,14 @@ def destroy_reported_item
     mark_as_reviewed
   end
 
+  def mark_as_nsfw
```

Comment:
How about either ignoring whiny requests, talking to the user or setting the profile NSFW (which would federate)? I'm really not a fan of your "well, I'll just change it locally and don't care about the rest of the world" solution here.